akka-cluster/src/main/scala/akka/cluster/Member.scala

```diff
@@ -195,26 +196,33 @@ object Member {
 
   /**
    * Picks the Member with the highest "priority" MemberStatus.
+   * Where highest priority is furthest along the membership state machine
+   * FIXME - include shutting down
    */
   def highestPriorityOf(m1: Member, m2: Member): Member = {
     if (m1.status == m2.status)
       // preserve the oldest in case of different upNumber
       if (m1.isOlderThan(m2)) m1 else m2
     else
       (m1.status, m2.status) match {
-        case (Removed, _)  => m1
-        case (_, Removed)  => m2
-        case (Down, _)     => m1
-        case (_, Down)     => m2
-        case (Exiting, _)  => m1
-        case (_, Exiting)  => m2
-        case (Leaving, _)  => m1
-        case (_, Leaving)  => m2
-        case (Joining, _)  => m2
-        case (_, Joining)  => m1
-        case (WeaklyUp, _) => m2
-        case (_, WeaklyUp) => m1
-        case (Up, Up)      => m1
+        case (Removed, _) => m1
+        case (_, Removed) => m2
+        case (PreparingForShutdown, _) =>
+          m1 // FIXME Consider moving these after down and exiting. Should we allow a node to still leave?
+        case (_, PreparingForShutdown) => m2
```

Comment:
I think those should be between WeaklyUp and Up.

Reply:
Agreed